activesupport/lib/active_support/message_verifier.rb

```diff
@@ -64,54 +66,162 @@ def valid_message?(signed_message)
     #
     # Returns +nil+ if the message is not Base64-encoded.
     #
-    #   invalid_message = "f--46a0120593880c733a53b6dad75b42ddc1c8996d"
+    #   invalid_message = "header.claims.MTkwMmI3MzY0MDA4NDM2MzVmYjgwMGEzOTJjOGQ2ZDM2Y2VhYjZiYg=="
     #   verifier.verified(invalid_message) # => nil
     #
     # Raises any error raised while decoding the signed message.
     #
-    #   incompatible_message = "test--dad7b06c94abba8d46a15fafaef56c327665d5ff"
-    #   verifier.verified(incompatible_message) # => TypeError: incompatible marshal file format
-    def verified(signed_message)
-      if valid_message?(signed_message)
-        begin
-          data = signed_message.split("--")[0]
-          @serializer.load(decode(data))
-        rescue ArgumentError => argument_error
-          return if argument_error.message =~ %r{invalid base64}
-          raise
-        end
+    #   incompatible_message = "aGVhZGVy.Y2xhaW1z.OWYzY2Y1Mzc1MmUxNjJlMzFhZDY3ODVlNGMyMTQ5YWE2OTc0NjQ5MQ=="
+    #   verifier.verified(incompatible_message) # raises TypeError: incompatible marshal file format
+    #
+    # Raises +InvalidClaims+ if the message doesn't have the same purpose as that
+    # given in the options.
+    #
+    #   verifier.verified(signed_message, for: 'something_else') # raises ActiveSupport::Claims::InvalidClaims
+    #
+    # Raises +ExpiredClaims+ if the message is expired.
+    #
+    #   expired_message = verifier.generate('a private message', expires_at: 1.day.ago)
+    #   verifier.verified(signed_message) # raises ActiveSupport::Claims::ExpiredClaims
```

Comment:
Oh shoot !! You are right. The whole purpose of ```verified ```was that it shouldn't raise anything. I forgot about the raise by ```Claims```.